pkg/apis/datavolumecontroller/v1alpha1/types.go

```diff
@@ -31,56 +32,79 @@ type DataVolume struct {
 	Status DataVolumeStatus `json:"status"`
 }
 
+// DataVolumeSpec defines our specification for a DataVolume type
 type DataVolumeSpec struct {
-	Source DataVolumeSource                  `json:"source"`
-	PVC    *corev1.PersistentVolumeClaimSpec `json:"pvc"`
+	//Source is the src of the data for the requested DataVolume
+	Source DataVolumeSource `json:"source"`
+	//PVC is a pointer to the PVC Spec we want to use
+	PVC *corev1.PersistentVolumeClaimSpec `json:"pvc"`
 }
 
+// DataVolumeSource represents the source for our Data Volume, this can be HTTP, S3 or an existing PVC
 type DataVolumeSource struct {
 	HTTP *DataVolumeSourceHTTP `json:"http,omitempty"`
 	S3   *DataVolumeSourceS3   `json:"s3,omitempty"`
 	PVC  *DataVolumeSourcePVC  `json:"pvc,omitempty"`
 }
 
+// DataVolumeSourcePVC provides the parameters to create a Data Volume from an existing PVC
 type DataVolumeSourcePVC struct {
 	Namespace string `json:"namespace,omitempty"`
 	Name      string `json:"name,omitempty"`
 }
 
+// DataVolumeSourceS3 provides the parameters to create a Data Volume from an S3 source
```

Comment:
DataVolumes also work with http sources, AFAIK.

Reply:
?  This comment is for a DataVolumeS3 struct?

Reply:
Nevermind... I didn't notice that.